pkg/entities/nfts.go

```diff
@@ -654,6 +655,41 @@ func (e *Entity) GetAllNFTSalesTracker() ([]response.NFTSalesTrackerResponse, er
 	return resp, nil
 }
 
+func (e *Entity) DeleteNFTSalesTracker(guildID, contractAddress string) error {
+	salesTracker, err := e.repo.NFTSalesTracker.GetNFTSalesTrackerByContractAndGuildID(guildID, contractAddress)
+	if err != nil {
+		e.log.Fields(logger.Fields{"guildID": guildID}).Error(err, "[repo.GuildConfigSalesTracker.GetByGuildID] failed to get nft sales trackers config")
+		return err
+	}
+	return e.repo.NFTSalesTracker.DeleteNFTSalesTracker(model.NFTSalesTracker{ID: salesTracker.ID})
+}
+
+func (e *Entity) GetNFTSaleSTrackerByGuildID(guildID string) (*response.NFTSalesTrackerGuildResponse, error) {
+	data, err := e.repo.NFTSalesTracker.GetSalesTrackerByGuildID(guildID)
+	if err != nil {
+		e.log.Fields(logger.Fields{"guildID": guildID}).Error(err, "[entity.GetNFTSaleSTrackerByGuildID] failed to get nft sales trackers")
+		return nil, err
+	}
+	if len(data) == 0 {
+		return nil, fmt.Errorf("Guild has no sales trackers")
+	}
+
+	collection := []response.NFTSalesTracker{}
+	for _, v := range data {
```

Comment:
i think can preload nft_sales_tracker

Reply:
use preload to query on database but it repeats guildID and channelID several times in collection so i use this loop to remove the unnecessary information

Reply:
unnecessary information will not be used by bot, i think use preload will save time here